api/src/main/java/com/github/streamshub/console/api/SchemasResource.java

```diff
@@ -0,0 +1,93 @@
+package com.github.streamshub.console.api;
+
+import java.util.Base64;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Optional;
+
+import jakarta.inject.Inject;
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.NotFoundException;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.PathParam;
+import jakarta.ws.rs.Produces;
+import jakarta.ws.rs.core.MediaType;
+import jakarta.ws.rs.core.Response;
+
+import org.eclipse.microprofile.openapi.annotations.media.Content;
+import org.eclipse.microprofile.openapi.annotations.parameters.Parameter;
+import org.eclipse.microprofile.openapi.annotations.responses.APIResponse;
+import org.eclipse.microprofile.openapi.annotations.tags.Tag;
+import org.jboss.logging.Logger;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.github.streamshub.console.api.support.KafkaContext;
+import com.github.streamshub.console.api.support.serdes.ArtifactReferences;
+import com.github.streamshub.console.api.support.serdes.MultiformatSchemaParser;
+
+import io.apicurio.registry.resolver.DefaultSchemaResolver;
+import io.apicurio.registry.resolver.SchemaResolver;
+import io.apicurio.registry.rest.client.RegistryClient;
+
+@Path("/api/schemas/{schemaId}")
+@Tag(name = "Schema Registry Resources")
+public class SchemasResource {
+
+    @Inject
+    Logger logger;
+
+    @Inject
+    ObjectMapper objectMapper;
+
+    @Inject
+    Map<String, KafkaContext> kafkaContexts;
+
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    @APIResponse(responseCode = "200", ref = "Configurations", content = @Content())
+    @APIResponse(responseCode = "404", ref = "NotFound")
+    @APIResponse(responseCode = "500", ref = "ServerError")
+    @APIResponse(responseCode = "504", ref = "ServerTimeout")
+    public Response getSchemaContent(
+            @Parameter(description = "Schema identifier")
+            @PathParam("schemaId")
+            String schemaId) {
+
+        int clusterTerm = schemaId.indexOf('.');
+
+        if (clusterTerm < 0) {
+            throw new NotFoundException("No such schema");
+        }
+
+        String clusterId = new String(Base64.getUrlDecoder().decode(schemaId.substring(0, clusterTerm)));
+
+        if (!kafkaContexts.containsKey(clusterId)) {
+            throw new NotFoundException("No such schema");
```

Comment:
This endpoint is meant to be non-hierarchical, with the Kafka cluster Id being part of the opaque schema Id. I'm thinking now that's confusing and doesn't really buy us anything, so I'll probably switch it to `/api/kafkas/{kafkaId}/schemas/{schemaId}`. Basically, my intent was to disassociate a Kafka instance from a schema since they aren't related in Apicurio. However, in the console we need the Kafka Id to find the Apicurio API URL/client.

Reply:
To answer your question, even though internally it can't find the Kafka cluster, from the perspective of a client to `api/schemas/{schemaId}`, a 404 means the schema wasn't found, regardless of the implementation of how to find it.

Reply:
I'm wondering whether the model of associating a schema registry to a kafka cluster is the right one.  In Kafka, other the client knows about the schema registry.  It is possible for topic A to use a schema hosted on one registry and topic B to use a schema on different.

I don't have a concrete suggestion to make right now but wanted to express unease. Did we research how other products/vendors approach this problem?  Are we going in the right direction on this?

I suppose we are talking about eventually having some kind of topic metadata service.  Once such a thing exists, it could identify the topic's schema and the registry that holds that schema.

Reply:
> Did we research how other products/vendors approach this problem?

Both AKHQ [1] and kafka-ui [2] do it per cluster. I agree that having the metadata storage for topics would make a difference in how we approach this. Associating them to Kafka clusters also doesn't feel ideal to me, but I wonder how often multiple registries are being used in an environment. The only way I can think to do that (without metadata storage) would be to allow for per-topic configuration of the registry, which could possibly be a future change if someone asks for it.

[1] https://akhq.io/docs/configuration/brokers.html
[2] https://docs.kafka-ui.provectus.io/configuration/serialization-serde#schemaregistry

Reply:
> Associating them to Kafka clusters also doesn't feel ideal to me, but I wonder how often multiple registries are being used in an environment. 

I don't know that either.

The approach you are taking seems like a reasonable place to start.

Reply:
@k-wall another option that might work is to allow for schema registries to be configured at the top level and then individual Kafka clusters could reference them. E.g.

```yaml
schemaRegistries:
  - name: production-registry
    url: http://prod.example.com/...
  - name: staging-registry
    url: http://staging.example.com...

kafka:
  clusters:
    - name: prod-k1
      schemaRegistry: production-registry
    - name: prod-k2
      schemaRegistry: production-registry
    - name: staging-k3
      schemaRegistry: staging-registry
```

This would make the `SchemasResource` URL a more related to reality: `/api/schema-registries/{registryName}/schemas/{schemaId}` and make the configuration a tiny bit DRYer. We're still binding a single registry per Kafka cluster, but the config for each registry is "reusable" within the config.

wdyt?

Reply:
I've gone ahead with this change. As mentioned, this will reduce duplicate configurations between Kafka clusters and also establishes a pattern that should be extended to Prometheus instance configurations and how they are bound to a Kafka cluster for metrics retrieval.